new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Remove magic and method_missing from Chef::Resource lookup
+
+This proposal:
+
+- Switches to explicit methods for Chef recipe DSL instead of method_missing
+- Removes all special treatment of classes in the Chef::Resource and Chef::Provider namespaces
+- Automatically adds resource DSL for all named descendants of Chef::Resource, no matter what their namespace
```

Comment:
Recording the name on `.inherited` is fine, as long as the system recognizes calling provides manually later and removes the very-briefly set incorrect name. As it stands, the NodeMap code doesn't do that.